src/web/page/mod.rs

```diff
@@ -3,14 +3,29 @@ pub(crate) mod web_page;
 
 pub(crate) use templates::TemplateData;
 
-use serde::Serialize;
+use crate::f_a::IconStr;
+use serde::ser::{Serialize, SerializeStruct, Serializer};
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
 pub(crate) struct GlobalAlert {
     pub(crate) url: &'static str,
     pub(crate) text: &'static str,
     pub(crate) css_class: &'static str,
-    pub(crate) fa_icon: &'static str,
+    pub(crate) fa_icon: crate::f_a::icons::IconTriangleExclamation,
+}
+
+impl Serialize for GlobalAlert {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = serializer.serialize_struct("GlobalAlert", 4)?;
+        s.serialize_field("url", &self.url)?;
+        s.serialize_field("text", &self.text)?;
+        s.serialize_field("css_class", &self.css_class)?;
+        s.serialize_field("fa_icon", &self.fa_icon.icon_name())?;
```

Comment:
the `.icon_name` is useless to render the icon, 
so we could also just `serde(skip)` the field and drop the custom `Serialize` impl again, right?

Reply:
:+1: